src/pages/Settings/Settings.js

```diff
@@ -1,146 +1,182 @@
+import { useQuery, useMutation } from '@apollo/react-hooks'
+import { isEmpty, capitalize } from 'lodash/fp'
+import './Settings.module.css'
+import { sliceHash as presentHash } from 'utils'
+import HashAvatar from 'components/HashAvatar'
+import Modal from 'components/Modal'
 import PrimaryLayout from 'components/layout/PrimaryLayout'
 import Button from 'components/Button'
-import Input from 'components/Input'
+import HposSettingsQuery from 'graphql/HposSettingsQuery.gql'
+import HposStatusQuery from 'graphql/HposStatusQuery.gql'
+import HposUpdateVersionMutation from 'graphql/HposUpdateVersionMutation.gql'
 
-import './Settings.module.css'
+export const createLabelfromSnakeCase = string => string.replace(/([A-Z])/g, ' $1').split(' ').map(word => capitalize(word)).join(' ')
+
+const DEFAULT_PORT_NAMES = ['Device Admin', 'HC Network', 'Hosting']
+const NOT_AVAILABLE = 'Not Available'
 
-const portValidationRules = yup.number()
-  .typeError('Port must be specified.') // TypeError because empty value gets cast into NaN
-  .min(1000, 'Ports must be between 1000 and 65000.')
-  .max(65000, 'Ports must be between 1000 and 65000.')
-  .required()
-const SettingsValidationSchema = yup.object().shape({
-  hostName: yup.string().required(),
-  hostPubKey: yup.string().required(),
-  registrationEmail: yup.string()
-    .email()
-    .required(),
-  deviceName: yup.string().required(),
-  networkId: yup.string().required(),
-  deviceAdminPort: portValidationRules,
-  hcAdminPort: portValidationRules,
-  hcNetworkPort: portValidationRules,
-  hostingPort: portValidationRules
-})
-
-export function Settings ({
-  settings,
-  updateSettings,
-  history: { push },
-  toggleSshAccess
-}) {
-  const { register, handleSubmit, errors } = useForm({
-    defaultValues: settings,
-    validationSchema: SettingsValidationSchema
+// Data - Mutation hook
+function useUpdateVersion () {
+  const [hposUpdateVersion] = useMutation(HposUpdateVersionMutation)
+  return (availableVersion) => hposUpdateVersion({
+    variables: { availableVersion }
   })
+}
 
-  const [sshAccessVal, setSshAccess] = useState(false)
-  const handleToggleSshAccess = (e) => {
-    e.preventDefault()
-    setSshAccess(e.target.checked)
-    toggleSshAccess()
-  }
+export function Settings ({ history: { push } }) {
+  const { data: { hposSettings: settings = [] } = {} } = useQuery(HposSettingsQuery)
+  const { data: { hposStatus: status = [] } = {} } = useQuery(HposStatusQuery)
+
+  const updateVersion = useUpdateVersion()
+  const [softwareUpdateVersion, setSoftwareUpdateVersion] = useState()
+  const showSoftwareUpdateModal = availableVersion => setSoftwareUpdateVersion(availableVersion)
 
-  const onSubmit = settings => {
-    updateSettings(settings)
+  let updateAvailable = false
+  let availableVersion, currentVersion
+  if (!isEmpty(status) && !isEmpty(status.versionInfo)) {
+    availableVersion = status.versionInfo.availableVersion
+    currentVersion = status.versionInfo.currentVersion
```

Comment:
you can use `lodash/fp/get` for this
```
const availableVersion = get('versionInfo.availableVersion', status)
```

Reply:
No problem; this is an easy update.  For clarity though, what is the reason for this preference?  As far as I can tell from the lodash docs, there is no additional error handling provided by this method (although there is a default state option for unresolved values that we could use). What added value does the lodash get method provide for us here?

Reply:
get doesn't throw if the attributes aren't present, it just returns a value (default `undefined` I think). So you can avoid the `isEmpty` checks, so

```
const availableVersion = get('versionInfo.availableVersion', status)
```

replaces lines 37-39

Reply:
Got it. :+1: